src/types.ts

```diff
@@ -224,6 +224,275 @@ export interface TransactionArguments {
     treatAsGrossAmount?: boolean;
     forceSweep?: boolean;
     feePayerInfo?: TransactionArgumentsFeePayerInfo;
+    travelRuleMessage?: TravelRule;
+}
+
+export type OwnershipProof = {
+    type: string;
+    proof: string;
+};
+
+export enum TravelRuleAddressTypeCode {
+    HOME = "HOME",
+    BIZZ = "BIZZ",
+    GEOG = "GEOG"
+}
+
+type TravelRulePersonAddress = {
+    addressType?: TravelRuleAddressTypeCode;
+    department?: string;
+    subDepartment?: string;
+    streetName?: string;
+    buildingNumber?: string;
+    buildingName?: string;
+    floor?: string;
+    postBox?: string;
+    room?: string;
+    postCode?: string;
+    townName?: string;
+    townLocationName?: string;
+    districtName?: string;
+    countrySubDivision?: string;
+    addressLine?: string[];
+    country?: string;
+};
+
+export interface ValidateTravelRuleVaspInfo {
+    transactionAsset: string;
+    destination: string;
+    transactionAmount: string;
+    originatorVASPdid: string;
+    originatorEqualsBeneficiary: boolean;
+    beneficiaryVASPdid?: string;
+    beneficiaryVASPname?: string;
+    beneficiaryName?: string;
+    beneficiaryAccountNumber?: string;
+    beneficiaryAddress?: TravelRulePersonAddress;
+    beneficiaryProof?: OwnershipProof;
+    travelRuleBehavior?: boolean;
+}
+
+export interface ValidateCreateTravelRuleTransaction {
+    transactionAsset: string;
+    transactionAmount: string;
+    originatorDid?: string;
+    beneficiaryDid?: string;
+    originatorVASPdid: string;
+    beneficiaryVASPdid?: string;
+    beneficiaryVASPname?: string;
+    transactionBlockchainInfo?: {
+        origin?: string;
+        destination?: string;
+        txHash?: string;
+    };
+    originator?: TROriginator;
+    beneficiary?: TROriginator;
+    pii?: PII;
+    pii_url?: string;
+    protocol?: ["TRLight", "TRP", "OpenVASP"];
```

Comment:
According to the docs it should be a string, did you mean: `"TRLight" | "TRP" | "OpenVASP"`

Reply:
Yes, god catch it is a string. Changed!